src/utils.jl

```diff
@@ -190,3 +191,20 @@ function _normal_quantiles_from_ranks!(q, r; α=3//8)
     q .= (r .- α) ./ (n - 2α + 1)
     return q
 end
+
+# utilities for supporting input arrays with an arbitrary number of dimensions
+
+_sample_dims(x::AbstractArray) = ntuple(identity, min(2, ndims(x)))
+
+_param_dims(x::AbstractArray) = ntuple(i -> i + 2, max(0, ndims(x) - 2))
+
+_param_axes(x::AbstractArray) = map(Base.Fix1(axes, x), _param_dims(x))
+
+function _params_array(x::AbstractArray, param_dim::Int=3)
+    sample_dims = ntuple(identity, param_dim - 1)
+    return reshape(x, map(Base.Fix1(size, x), sample_dims)..., :)
```

Comment:
I prefer not having the conditional check, since we know the `Colon` should be last. I did change it to avoid mapping over the tuple though and instead construct it directly.